service.py

```diff
@@ -64,18 +64,23 @@ def org_user(org):
                 logger.info('User not part of organization')
                 logger.info(f'Trying to add user: {invi_username}')
                 invi_response = requests.put(f"{config.github_base_url}/orgs/{org}/memberships/{invi_username}", auth=(username, token))
-                if invi_response.status_code == 200:
-                    decoded_data = json.loads(invi_response.content.decode('utf-8-sig'))
-                    if decoded_data.get('state') == "pending":
-                        logger.info(f"Organization invitation sent to username: {invi_username}")
-                    if decoded_data.get('state') == "active":
-                        logger.info(f"User with username: {invi_username}, already part of organization")
+                if invi_response.ok:
+                    if invi_response.status_code == 200:
+                        decoded_data = json.loads(invi_response.content.decode('utf-8-sig'))
+                        if decoded_data.get('state') == "pending":
+                            logger.info(f"Organization invitation sent to username: {invi_username}")
+                        if decoded_data.get('state') == "active":
+                            logger.info(f"User with username: {invi_username}, already part of organization")
+                    else:
+                        logger.warning(f"User with username: {invi_username} invited, but unexpected response with status code: {invi_response.status_code}")
+                else:
+                    if invi_response.status_code == 422 or invi_response.status_code == 403:
+                        logger.warning(f"Organization invitation could not be sent to username: {invi_username}")
+                        logger.warning(f"Failed with error code: {invi_response.status_code}")
+                    else:
```

Comment:
Approving with a suggestion: 'If' and 'else' block here can be unified without checking any specific code. As long as it is non-ok response, it doesn't matter what code it is.

Reply:
Guessing the original developer had some intention behind it so I thought I'll leave it as is